src/uct/api/uct.h

```diff
@@ -259,6 +272,24 @@ typedef struct uct_component_attr {
      */
     uct_md_resource_desc_t *md_resources;
 
+    /** Number of connection manager resources */
+    unsigned               cm_resource_count;
+
+    /**
+     * Array of connection manager resources. When used, it should be initialized
+     * prior to calling @ref uct_component_query with a pointer to an array,
+     * which is large enough to hold all connection manager resource entries.
+     * After the call, this array will be filled with information about existing
+     * connection manager resources.
+     * In order to allocate this array, you can call @ref uct_component_query
+     * twice: The first time would only obtain the amount of entries required,
+     * by specifying @ref UCT_COMPONENT_ATTR_FIELD_CM_RESOURCE_COUNT in
+     * field_mask. Then the array could be allocated with the returned number of
+     * entries, and passed to a second call to @ref uct_component_query, this
+     * time setting field_mask to @ref UCT_COMPONENT_ATTR_FIELD_CM_RESOURCES.
+     */
+    uct_cm_resource_desc_t *cm_resources;
```

Comment:
how it can be > 1?

Reply:
it can't be but it can be 0. i'll rename to cm_resource and update the doc.